contracts/src/token_locker.cairo

```diff
@@ -0,0 +1,177 @@
+use starknet::ContractAddress;
+
+#[starknet::interface]
+trait ITokenLocker<TContractState> {
+    fn lock(ref self: TContractState, token_contract: ContractAddress, amount: u256);
+    fn unlock(ref self: TContractState, token_contract: ContractAddress, lock_time: u64);
+    fn get_locked_amount(
+        self: @TContractState,
+        token_contract: ContractAddress,
+        owner: ContractAddress,
+        lock_time: u64
+    ) -> u256;
+    fn get_time_left(
+        self: @TContractState,
+        token_contract: ContractAddress,
+        owner: ContractAddress,
+        lock_time: u64
+    ) -> u64;
+}
+
+#[starknet::contract]
+mod TokenLocker {
+    use openzeppelin::token::erc20::{ERC20ABIDispatcher, ERC20ABIDispatcherTrait};
+    use starknet::{ContractAddress, get_caller_address, get_contract_address, get_block_timestamp};
+    use unruggable::errors::{
+        ZERO_LOCK, ZERO_LOCK_TIME, LOCK_EXIST, TRANSFER_FAIL, LOCK_NONEXIST, STILL_LOCKED
+    };
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        TokenLocked: TokenLocked,
+        TokenUnlocked: TokenUnlocked
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct TokenLocked {
+        locker: ContractAddress,
+        token: ContractAddress,
+        lock_time: u64,
```

Comment:
```suggestion
        lock_time: u64,
```

you used `lock_time` here, the same variable name as the `lock_time` in the Storage struct which refers to how long a token is going to be locked. 

I think renaming it to `lock_timestamp` or would be better for clarity. same for `unlock_time`

Reply:
done